central/networkgraph/service/service_impl_postgres_test.go

```diff
@@ -0,0 +1,517 @@
+	s.NoError(err)
+
+	s.entityDataStore, err = networkEntityDS.GetTestPostgresDataStore(s.T(), db.DB)
+	s.NoError(err)
+
+	s.flowDataStore, err = networkFlowDS.GetTestPostgresClusterDataStore(s.T(), db.DB)
+	s.NoError(err)
+
+	s.policyDataStore, err = networkPolicyDS.GetTestPostgresDataStore(s.T(), db.DB)
+	s.NoError(err)
+
+	s.configDataStore, err = configDS.GetTestPostgresDataStore(s.T(), db.DB)
+	s.NoError(err)
+
+	s.treeMgr = networktree.Singleton()
+
+	s.service = New(
+		s.flowDataStore,
+		s.entityDataStore,
+		s.treeMgr,
+		s.deploymentsDataStore,
+		s.clusterDataStore,
+		s.policyDataStore,
+		s.configDataStore,
+	)
+
+	s.db = db
+}
+
+func (s *networkGraphServiceSuite) TeardownTest() {
+	s.db.Teardown(s.T())
+}
+
+func externalFlow(deployment *storage.Deployment, entity *storage.NetworkEntity, ingress bool) *storage.NetworkFlow {
+	deploymentEntityInfo := &storage.NetworkEntityInfo{
+		Type: storage.NetworkEntityInfo_DEPLOYMENT,
+		Id:   deployment.Id,
+		Desc: &storage.NetworkEntityInfo_Deployment_{
+			Deployment: &storage.NetworkEntityInfo_Deployment{
+				Namespace: deployment.Namespace,
+			},
+		},
+	}
+
+	entityInfo := entity.GetInfo()
+
+	if ingress {
+		return &storage.NetworkFlow{
+			Props: &storage.NetworkFlowProperties{
+				SrcEntity:  entityInfo,
+				DstEntity:  deploymentEntityInfo,
+				DstPort:    1234,
+				L4Protocol: storage.L4Protocol_L4_PROTOCOL_TCP,
+			},
+			LastSeenTimestamp: nil,
+			ClusterId:         deployment.ClusterId,
+		}
+	}
+	return &storage.NetworkFlow{
+		Props: &storage.NetworkFlowProperties{
+			DstEntity:  entityInfo,
+			SrcEntity:  deploymentEntityInfo,
+			DstPort:    1234,
+			L4Protocol: storage.L4Protocol_L4_PROTOCOL_TCP,
+		},
+		LastSeenTimestamp: nil,
+		ClusterId:         deployment.ClusterId,
+	}
+}
+
+func (s *networkGraphServiceSuite) TestGetExternalNetworkFlows() {
+	ctx := sac.WithGlobalAccessScopeChecker(
+		context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Deployment, resources.NetworkGraph),
+		),
+	)
+
+	globalWriteAccessCtx := sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.NetworkGraph, resources.Deployment)))
+
+	entityID, _ := externalsrcs.NewClusterScopedID(testCluster, "192.168.1.1/32")
+	entityID2, _ := externalsrcs.NewClusterScopedID(testCluster, "10.0.0.2/32")
+	entityID3, _ := externalsrcs.NewClusterScopedID(testCluster, "1.1.1.1/32")
+
+	entity := testutils.GetExtSrcNetworkEntity(entityID.String(), "ext1", "192.168.1.1/32", false, testCluster)
+	entity2 := testutils.GetExtSrcNetworkEntity(entityID2.String(), "ext2", "10.0.0.2/32", false, testCluster)
+	entity3 := testutils.GetExtSrcNetworkEntity(entityID3.String(), "ext3", "10.0.100.25/32", false, testCluster)
```

Comment:
I think arrays might be more readable here, you can always define a pointer to one of the elements later on if you think that would make the code easier to read, but having `entity2` or `entity[1]` doesn't really make much difference IMO.

Reply:
I just noticed right below this comment there's `entities := []*storage.NetworkEntity`, so even more reason to directly create the array.